### ISSUE TITLE
Compile ompi with UCX--error: spml_ucx_component.c:122:20: error: too many arguments to function 'ucp_init'

### ISSUE THREAD

**Issue by Jack00112** [openucx/ompi-mirror#4]
**Env:**
OS: Ubuntu 14.04
gcc version: gcc (Ubuntu 4.8.4-2ubuntu1~14.04.3) 4.8.4

**Step to reproduce:**
build ucx
# checkout Pull Request
% git clone https://github.com/openucx/ompi-mirror
% cd ompi-mirror
% git fetch origin pull/1/head:pr-1
% git checkout pr-1

$ cd ompi-mirror
$ ./autogen.pl
Configure with UCX:
$ mkdir build
$ cd build
../configure --prefix=/your_install_path/ --with-ucx=/path_to_ucx_installation
Build:
$ make
$ make install

During run command "make", the error appeared as below:
make[2]: Entering directory `/home/jacui/FM/newompi/ompi-mirror/build/oshmem/mca/spml/ucx'
  CC       spml_ucx_component.lo
../../../../../oshmem/mca/spml/ucx/spml_ucx_component.c: In function 'mca_spml_ucx_component_open':
../../../../../oshmem/mca/spml/ucx/spml_ucx_component.c:121:36: error: 'UCP_FEATURE_AMO' undeclared (first use in this function)
     err = ucp_init(UCP_FEATURE_RMA|UCP_FEATURE_AMO, 0, ucp_config,
                                    ^
../../../../../oshmem/mca/spml/ucx/spml_ucx_component.c:121:36: note: each undeclared identifier is reported only once for each function it appears in
../../../../../oshmem/mca/spml/ucx/spml_ucx_component.c:122:20: warning: passing argument 3 of 'ucp_init' from incompatible pointer type [enabled by default]
                    &mca_spml_ucx.ucp_context);
                    ^
In file included from ../../../../../oshmem/mca/spml/ucx/spml_ucx.h:28:0,
                 from ../../../../../oshmem/mca/spml/ucx/spml_ucx_component.c:23:
/home/jacui/FM/newUCX/ucx/install-debug/include/ucp/api/ucp.h:494:28: note: expected 'struct ucp_context **' but argument is of type 'struct ucp_config_t *'
 static inline ucs_status_t ucp_init(const ucp_params_t *params,
                            ^
../../../../../oshmem/mca/spml/ucx/spml_ucx_component.c:122:20: error: too many arguments to function 'ucp_init'
                    &mca_spml_ucx.ucp_context);
                    ^
In file included from ../../../../../oshmem/mca/spml/ucx/spml_ucx.h:28:0,
                 from ../../../../../oshmem/mca/spml/ucx/spml_ucx_component.c:23:
/home/jacui/FM/newUCX/ucx/install-debug/include/ucp/api/ucp.h:494:28: note: declared here
 static inline ucs_status_t ucp_init(const ucp_params_t *params,
                            ^
make[2]: *** [spml_ucx_component.lo] Error 1
make[2]: Leaving directory `/home/jacui/FM/newompi/ompi-mirror/build/oshmem/mca/spml/ucx'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/jacui/FM/newompi/ompi-mirror/build/oshmem'
make: *** [all-recursive] Error 1

**Comment by miked-mellanox** [comment]
@Jack00112 - upstream OMPI should be used instead of mirror.

https://github.com/open-mpi/ompi-release/

@yosefe , @shamisp - probably we should remove ompi-mirror as it is outdated.

**Comment by Jack00112** [comment]
@miked-mellanox , thank you for your reply, the old errors have disappeared, now new issue was shown as below:
ompi-release/build/ompi/.libs/libmpi.so: undefined reference to `opal_timer_base_get_usec'
collect2: error: ld returned 1 exit status

**Comment by miked-mellanox** [comment]
does it happen w/o --with-ucx as well? seems your src folder is tainted with mix of different builds.

**Comment by Jack00112** [comment]
@miked-mellanox , thank you for your reply, without --with-ucx, the error is the same.
ompi-release/build/ompi/.libs/libmpi.so: undefined reference to `opal_timer_base_get_usec'
collect2: error: ld returned 1 exit status

**Comment by miked-mellanox** [comment]
plz try build after fresh clone of ompi, if problem persist - plz open ompi ticket in ompi repo.

**Comment by Jack00112** [comment]
@miked-mellanox , thank you. Please ignore the latest reply, now I can compile successfully without  --with-ucx

**Comment by miked-mellanox** [comment]
cool, what was an issue?

**Comment by Jack00112** [comment]
@miked-mellanox , it's ok with --with-ucx, I just delete all the ompi source code and resync it again. Thank you very much.

**Comment by shamisp** [comment]
@miked-mellanox  Since the code was committed upstream I would suggest to remove the mirror
